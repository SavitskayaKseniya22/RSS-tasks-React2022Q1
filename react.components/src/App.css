.App {
  max-width: 100vw;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  justify-content: space-between;
}

* {
  box-sizing: border-box;
}

ul {
  list-style: none;
  margin: 0;
}
