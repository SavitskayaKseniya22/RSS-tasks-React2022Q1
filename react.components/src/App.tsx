import { BrowserRouter, Outlet } from 'react-router-dom';
import './App.css';
import { Footer } from './components/Footer';
import { Header } from './components/Header';

function App() {
  return (
    <div className="App" data-testid="app">
      <BrowserRouter>
        <Header />
      </BrowserRouter>
      <Footer />
      <Outlet />
    </div>
  );
}

export default App;
