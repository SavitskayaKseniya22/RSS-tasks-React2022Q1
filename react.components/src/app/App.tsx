import { Navigate, Outlet, Route, Routes } from 'react-router-dom';
import './App.css';
import { MainPage } from '../pages/mainPage/MainPage';
import { MyAds } from '../pages/myAds/MyAds';
import { AboutUs } from '../pages/aboutUs/AboutUs';
import { Header } from '../components/header/Header';
import { Footer } from '../components/footer/Footer';
import { Dispatch, useReducer } from 'react';
import { MainPageType, ReducerTypes, SearchItemDetailType } from '../interfaces';
import React from 'react';
import { SearchItemDetails } from '../components/searchItemDetails/SearchItemDetails';

const initialValues: MainPageType = {
  value: window.localStorage.getItem('searchValue') || '',
  response: [],
  isDownloading: false,
  isSearchOver: false,
  isError: false,
  activeCard: undefined,
  sort: 'latest',
  perPage: '20',
  pageNumber: '1',
  pageRange: 10,
};

export const ContextApp = React.createContext<{
  state: MainPageType;
  dispatch: Dispatch<ReducerTypes>;
}>({
  state: initialValues,
  dispatch: () => null,
});

function App() {
  const [state, dispatch] = useReducer(reducer, initialValues);

  function reducer(state: MainPageType, action: ReducerTypes): MainPageType {
    switch (action.type) {
      case 'handleDownload':
        return {
          ...state,
          response: action.payload.response,
          isDownloading: action.payload.load,
          isSearchOver: !action.payload.load,
          isError: action.payload.error || false,
        };

      case 'toggleCard':
        return { ...state, activeCard: action.payload.activeCard };

      case 'handleChange':
        return { ...state, value: action.payload.value };

      case 'handleSort':
        return { ...state, sort: action.payload.sort };
      case 'handlePerPage':
        return { ...state, perPage: action.payload.perPage };
      case 'handlePageNumber':
        return { ...state, pageNumber: action.payload.pageNumber };
      case 'handlePageRange':
        return { ...state, pageRange: action.payload.pageRange };
      /*
      case 'handleSearchForm':
        console.log(action.payload);
        console.log(state);
        return {
          ...state,
          sort: action.payload.sort,
          perPage: action.payload.perPage,
          pageNumber: action.payload.pageNumber,
          pageRange: action.payload.pageRange,
        };*/

      default:
        return state;
    }
  }

  return (
    <div className="App" data-testid="app">
      <ContextApp.Provider value={{ state, dispatch }}>
        <Header />
        <Footer />
        <Outlet />
        <Routes>
          <Route path="/my-ads" element={<MyAds />} />
          <Route path="/" element={<MainPage />} />
          <Route path="about-us" element={<AboutUs />} />
          <Route
            path="/card"
            element={state.activeCard ? <SearchItemDetails /> : <Navigate to="/" />}
          />

          <Route
            path="*"
            element={
              <div className="empty-page">
                <span>404</span>
              </div>
            }
          />
        </Routes>
      </ContextApp.Provider>
    </div>
  );
}

export default App;
