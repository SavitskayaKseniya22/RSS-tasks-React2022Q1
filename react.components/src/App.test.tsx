import { render, screen } from '@testing-library/react';
import App from './App';

test('check app page appearance', () => {
  render(<App />);
  const app = screen.getByTestId('app');
  expect(app).toMatchSnapshot();
});
