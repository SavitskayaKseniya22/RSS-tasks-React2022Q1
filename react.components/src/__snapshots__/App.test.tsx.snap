// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check app page appearance 1`] = `
<div
  class="App"
  data-testid="app"
>
  <header
    class="Header"
    data-testid="header"
  >
    <a
      aria-current="page"
      class="active-link"
      data-testid="main-page__link"
      href="/"
    >
      main page
    </a>
    <a
      class="inActiveLink"
      data-testid="about-us__link"
      href="/about-us"
    >
      about us
    </a>
  </header>
  <footer
    class="Footer"
    data-testid="footer"
  >
    <div>
      <a
        href="https://rs.school/js/"
        rel="noreferrer"
        target="_blank"
      >
        <img
          alt="logo"
          src="rs-school-js.svg"
          width="100"
        />
      </a>
    </div>
    <div>
      <a
        href="https://github.com/SavitskayaKseniya22"
        rel="noreferrer"
        target="_blank"
      >
        made by Kseniya Savitskaya
      </a>
    </div>
    <div>
      Â© 2022
    </div>
  </footer>
</div>
`;
